src/domain-engine.js

```diff
@@ -1,16 +1,16 @@
 import MetaDomain from './meta-domain'
 import getRegistration from './get-registration'
 import { get, set, createOrClone } from './utils'
-import { castPath } from './key-path'
+import { castPath, getKeyString } from './key-path'
 
 class DomainEngine {
   /**
    * @param {Microcosm} repo
    */
   constructor(repo) {
-    this.repo = repo
-    this.domains = []
     this.registry = {}
+    this.repo = repo
+    this.domains = [[[], this.repo]]
```

Comment:
Yeah. This is gross a bit. It would be great to just use a `Map` here, even if we wrote a light version for the sake of this project.